### PR TITLE
Add logic to track cloned entries

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses :
- Clone issue :  https://github.com/idmc-labs/Helix2.0/issues/188#issue-946419723
- Glide Number: https://github.com/idmc-labs/Helix2.0/issues/209#issue-954635249
- Age/Sex: https://github.com/idmc-labs/Helix2.0/issues/213#issue-956882035

## Changes
- Add logic to track the cloned entries efficiently, either by opening a new tab or by providing cloned entries location info
- Change the namespace "Glide Number" to "Event ID"
- Change the namespace "Age/Sex" to "Age & Gender"

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/views/Entry/EntryForm/index.tsx [hunk 1]
```diff
@@ -915,9 +929,19 @@ function EntryForm(props: EntryFormProps) {
                     <EntryCloneForm
                         entryId={entryIdToClone}
                         onCloseForm={hideEventCloneModal}
+                        showCloneAlert={showCloneAlert}
+                        setCloneEvents={setCloneEvents}
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/index.tsx:933]: The prop names should be renamed

File: src/views/Entry/EntryForm/EntryCloneForm/index.tsx [hunk 2]
```diff
@@ -148,7 +151,10 @@ function EventCloneForm(props: EventCloneFormProps) {
                     notifyGQLError(errors);
                 }
                 if (ok) {
+                    setCloneEvents(eventOptions);
+                    showCloneAlert();
                     notify({ children: 'Entry cloned successfully!' });
+
                     onPristineSet(true);
                     onCloseForm();
```
Comment (tnagorra) [inline src/views/Entry/EntryForm/EntryCloneForm/index.tsx:159]: Too many callbacks are called

File: src/components/Alert/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                            >
+                                {okayLabel}
+                            </Button>
+                        )}
+
+                >
+                    {
+                        cloneEvents && cloneEvents?.length < 5
+                            ? 'Would you like to open the cloned entries in new tab? You can also find them in Dashboard Entries'
+                            : 'Please check Dashboard for the cloned entries !'
```
Comment (tnagorra) [inline src/components/Alert/index.tsx:111]:
```suggestion
                            ? 'Would you like to open the cloned entries in new tab? You can also find them in the extraction page.'
                            : 'Please check the extraction page for the cloned entries!'
```

File: src/components/Alert/index.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                                    autoFocus
+                                >
+                                    {cancelLabel}
+                                </Button>
+                                <Button
+                                    className={_cs(styles.actionButton, okayButtonClassName)}
+                                    name="confirm-button"
+                                    onClick={handleAlertAction}
+                                    variant="primary"
+                                    autoFocus
```
Comment (tnagorra) [inline src/components/Alert/index.tsx:90]: There should only be one autoFocus element

File: src/components/Alert/index.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    const [showAlertModal, setShowAlertModal] = useState(false);
+
+    const handleConfirmModalShow = useCallback(() => {
+        setShowAlertModal(true);
+    }, []);
+
+    useEffect(() => {
+        handleConfirmModalShow();
+    }, [handleConfirmModalShow]);
```
Comment (tnagorra) [inline src/components/Alert/index.tsx:41]: This doesn't look correct.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Entry/EntryForm/EntryCloneForm/index.tsx (hunk 2 above)
Comment (priyesh777) [reply src/views/Entry/EntryForm/EntryCloneForm/index.tsx:159]: Unfortunately, we need all those callbacks

Comment (tnagorra) [reply src/views/Entry/EntryForm/EntryCloneForm/index.tsx]: No, we do not need them.

**Pull request merged**